architecture/2025-02-application-autoscaling.md

```diff
@@ -0,0 +1,239 @@
+        minReplicas: 1
+        maxReplicas: 10
+        cpuUtilization: 50
+        memoryLimit: 50
+        httpConcurrency: 50
+        }
+      }
+    ]
+  }
+}
+```
+
+Pros: consistent with existing manual scaling config
+
+Cons: not straightforward or intuitive as user doesn't know which autoscale config is required for the platform ? needs a platform specific discriminator
+May be, this is not a developer problem and should be handled by the platform operator. 
+
+#### Option 3: Adding it as top level common property
+
+```bicep
+resource container 'Applications.Containers@2023-10-01-preview' = {
+  name: 'myapp'
+  properties: {
+    image: 'myregistry.azurecr.io/myapp:latest'
+    autoscaling: {
+      minReplicas: 1
+      maxReplicas: 10
+      hpa: {
+        cpuUtilization: 50
+        memoryLimit: 50
+      }
+      http: {
+        httpConcurrency: 50
+      }  
+    }
+  }
+}
+``` 
+
+Cons:same as above, not straightforward or intuitive as user doesn't know which autoscale config is required for the platform ? May be, this is not a developer problem and should be handled by the platform operator.
+
+### Configuring Autoscaling in Radius Environment
+
+A platform engineer or the operator sets up the Radius environment for the applications to be deployed. Today Radius environment contains the configurations for the container runtime, identity provider and the secrets store. Since scaling is a runtime concern, the platform operator would also need an ability to configure the scaling policies in the environment configuration.
+
+```bicep
+resource environment 'Applications.Core/environments@2023-10-01-preview' = {
+  name: 'myenv'
+  properties: {
+    compute: {
+      kind: 'kubernetes'
+      namespace: 'default' 
+      identity: {          
+        kind: 'azure.com.workload'
+        oidcIssuer: oidcIssuer
+      }
+    autoscaling: {
+        minReplicas: 1
+        maxReplicas: 10
+        cpuUtilization: 50
+        memoryLimit: 50
+      }
+    }
+}
+```
+
+Pros:
+1. Autoscaling is an infrastructure problem, the platform engineer has the ability to configure the autoscaling policies in the environment configuration thus separating the concerns of the platform engineer and the developer. 
+1. Dev, test environments wouldn't require autoscaling policies and the platform engineer has more flexibility to configure the scaling policies based on the environment.
+1. The platform engineer can configure the scaling policies once and all the applications deployed in the environment will inherit the scaling policies.
+1. Developer can focus on the application modelling and not worry about the scaling policies. If needed the developer can override the scaling policies in the application resource definition. Overriding scenario could be for example, if the developer wants to scale based on some events from the Kafka or RabbitMQ modelled in the application definition 
+
+### Configuring Autoscaling as a core resource type
```

Comment:
I don't follow this completely - I see Radius resources as objects it needs to provision or create at deploy time, but autoscaling is not an object, but rather a configuration. I think this kind of pattern might be better handled through bicep params instead?